File: Java/Tema6_Ejercicio2.java
/*
 * TEMA 6: CADENAS DE CARACTERES - EJERCICIO 2
 * ============================================
 * Escriba un programa que convierta una cadena a mayúsculas y minúsculas.
 */

import java.util.Scanner;

public class Tema6_Ejercicio2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite una cadena: ");
        String cadena = scanner.nextLine();
        
        System.out.println("\n=== CONVERSIÓN DE CADENA ===");
        System.out.println("Original: " + cadena);
        System.out.println("Mayúsculas: " + cadena.toUpperCase());
        System.out.println("Minúsculas: " + cadena.toLowerCase());
        System.out.println("Primera letra mayúscula: " + cadena.substring(0, 1).toUpperCase() + 
                         cadena.substring(1).toLowerCase());
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite una cadena: JaVa ProGraMinG
 * 
 * === CONVERSIÓN DE CADENA ===
 * Original: JaVa ProGraMinG
 * Mayúsculas: JAVA PROGRAMING
 * Minúsculas: java programing
 * Primera letra mayúscula: Java programing
 */

File: JS/Tema6_Ejercicio2.js
/*
 * TEMA 6: CADENAS DE CARACTERES - EJERCICIO 2
 * ============================================
 * Conversión de mayúsculas/minúsculas.
 * 
 * EXPLICACIÓN:
 * - toUpperCase(): convertir a mayúsculas
 * - toLowerCase(): convertir a minúsculas
 * - Casos de uso comunes
 */

const texto = "Desarrollo En JavaScript";

console.log("=== CONVERSIÓN DE CASO ===");
console.log(`Original: "${texto}"`);
console.log(`Mayúsculas: "${texto.toUpperCase()}"`);
console.log(`Minúsculas: "${texto.toLowerCase()}"`);

// Primera letra mayúscula
function capitalizarPrimera(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
}

// Capitalize cada palabra
function capitalizarPalabras(str) {
    return str.split(' ').map(palabra => capitalizarPrimera(palabra)).join(' ');
}

console.log(`\nPrimera letra mayúscula: "${capitalizarPrimera(texto.toLowerCase())}"`);
console.log(`Cada palabra capitalizada: "${capitalizarPalabras(texto.toLowerCase())}"`);

/*
 * SALIDA ESPERADA:
 * === CONVERSIÓN DE CASO ===
 * Original: "Desarrollo En JavaScript"
 * Mayúsculas: "DESARROLLO EN JAVASCRIPT"
 * Minúsculas: "desarrollo en javascript"
 * 
 * Primera letra mayúscula: "Desarrollo en javascript"
 * Cada palabra capitalizada: "Desarrollo En Javascript"
 */
